### PR TITLE
add normalizer that leaved conjugation

### PR DESCRIPTION
**Pull request by katsutan**

### CODE REVIEW

File: sudachitra/normalizer_leaved_conjugation.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+class NormalizerLeavedConjugation:
+    def __init__(self, inflectoin_table_path, conjugation_type_table_path) -> None:
+        with open(inflectoin_table_path) as jf:
+            self.infl_table = json.load(jf)
+        with open(conjugation_type_table_path) as jf:
+            self.conj_type_table = json.load(jf)
+        self.errlog = open("err.log", "w")
+
+    def _read_pos(self, morpheme: Morpheme) -> list:
+        pos = list(morpheme.part_of_speech())
```
Comment (eiennohito) [inline sudachitra/normalizer_leaved_conjugation.py:17]: you don't need list here, tuple will work perfectly well

**Review by eiennohito — CHANGES_REQUESTED** [review]
1. Using string concatenation in the hot path is very bad for execution speed, do not do it. Use tuples for keys instead.
2. Call Morpheme methods only once. (this should be fixed eventually in SudachiPy, but for the time being)
3. Try to use 0.6.2 and its PosMatcher infrastructure, prefer using `get_part_of_speech_id()` instead of `get_part_of_speech()`
4. Tables should not be jsons, they should be Python files. Generating those Python files from JSON is OK.
5. Organize filters in multiple levels, first level should do simple blacklist lookup, second level should

**Comment by eiennohito** [comment]
Also, typo: inflectoin_table.json  -> inflection_table.json

File: sudachitra/normalizer_leaved_conjugation.py [hunk 2]
```diff
@@ -1,43 +1,56 @@
...
+            conj_form = pos[5].split("-")[0] + "-一般"
+        return (*pos[:4], conj_type, conj_form)
+    
+    def _is_changed_conjugation_type(self, key) -> bool:
+        return key in self.conj_type_table
 
     def normalized(self, morpheme: Morpheme) -> str:
         normalized_token = morpheme.normalized_form()
-        if morpheme.part_of_speech()[0] not in CONJUGATIVE_POS:
+        if not self.is_conjugative_pos(morpheme) or not self.is_needs_inflection(morpheme):
```
Comment (eiennohito) [inline sudachitra/normalizer_leaved_conjugation.py:37]: this should be one operation instead of two

**Review by eiennohito — APPROVED** [review]
LGTM, thanks for fixes

**Comment by t-yamamura** [comment]
It would be helpful to have some more comments and documents on `NormalizerLeavedConjugation`.
In particular, since `NormalizerLeavedConjugation` is a WORD_FORM_TYPE that does not exist in SudachiPy, so it helps understanding to explain how it differs from the usual normalization process.

**Pull request merged**